lib/rex/proto/ms_dtyp.rb

```diff
@@ -261,6 +266,404 @@ class MsDtypSecurityDescriptor < BinData::Record
+            ace.body.access_mask.wd = true
+            ace.body.access_mask.wo = true
+          when 'KR'
+            ace.body.access_mask.protocol |= 0x19
+          when 'KW'
+            ace.body.access_mask.protocol |= 0x06
+          when 'KX'
+            ace.body.access_mask.protocol |= 0x19
+          when 'NR', 'NW', 'NX'
+            raise RuntimeError.new('SDDL parse error on unsupported ACE access right: ' + right)
+          when ''
+          else
+            raise RuntimeError.new('SDDL parse error on unknown ACE access right: ' + right)
+          end
+        end
+
+        unless object_guid.blank?
+          begin
+            guid = MsDtypGuid.new(object_guid)
+          rescue StandardError
+            raise RuntimeError.new('SDDL parse error on invalid object GUID: ' + object_guid)
+          end
+
+          unless ace.body.respond_to?('object_type=')
+            raise RuntimeError.new('SDDL error on setting object type for incompatible ACE type')
+          end
+          ace.body.flags.ace_object_type_present = true
+          ace.body.object_type = guid
+        end
+
+        unless inherit_object_guid.blank?
+          begin
+            guid = MsDtypGuid.new(inherit_object_guid)
+          rescue StandardError
+            raise RuntimeError.new('SDDL parse error on invalid object GUID: ' + inherit_object_guid)
+          end
+
+          unless ace.body.respond_to?('inherited_object_type=')
+            raise RuntimeError.new('SDDL error on setting object type for incompatible ACE type')
+          end
+          ace.body.flags.ace_inherited_object_type_present = true
+          ace.body.inherited_object_type = guid
+        end
+
+        unless account_sid.blank?
+          ace.body.sid = self.parse_sddl_sid(account_sid, domain_sid: domain_sid)
+        end
+
+        unless resource_attribute.blank?
+          raise RuntimeError.new('SDDL parse error on unsupported resource attribute: ' + resource_attribute)
+        end
+
+        ace
+      end
+
+      def parse_sddl_aces(aces, domain_sid:)
+        ace_regex = /\([^\)]*\)/
+
+        invalid_aces = aces.split(ace_regex).reject(&:empty?)
+        unless invalid_aces.empty?
+          raise RuntimeError.new('SDDL parse error on malformed ACE: ' + invalid_aces.first)
+        end
+
+        aces.scan(ace_regex).map do |ace_text|
+          self.parse_sddl_ace(ace_text[1...-1], domain_sid: domain_sid)
+        end
+      end
+
+      def parse_sddl_sid(sid, domain_sid:)
+        # see: https://learn.microsoft.com/en-us/windows/win32/secauthz/sid-strings
+
+        # these can be validated using powershell where ?? is the code
+        #   (ConvertFrom-SddlString -Sddl "O:??").RawDescriptor.Owner
+        case sid
+        when 'AA'  # SDDL_ACCESS_CONTROL_ASSISTANCE_OPS
+          sid = Rex::Proto::Secauthz::WellKnownSids::DOMAIN_ALIAS_SID_ACCESS_CONTROL_ASSISTANCE_OPS
+        when 'AC'  # SDDL_ALL_APP_PACKAGES
+          sid = 'S-1-15-2-1'.freeze
```

Comment:
> 2.4.2.4 Well-Known SID Structures

Also under here:
https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/81d92bba-d22b-4a8c-908a-554ab29148ab

Reply:
I'll add it to the definitions as `SECURITY_ALL_APP_PACKAGES`